crypto/bls/blst/public_key.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2023 The klaytn Authors
+// This file is part of the klaytn library.
+//
+// The klaytn library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The klaytn library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
+
+package blst
+
+import (
+	"github.com/klaytn/klaytn/crypto/bls/types"
+)
+
+type publicKey struct {
+	// Pointer to underlying blst struct, hence the name 'p'
+	p *blstPublicKey
+}
+
+func PublicKeyFromBytes(b []byte) (types.PublicKey, error) {
+	if len(b) != types.PublicKeyLength {
+		return nil, types.ErrPublicKeyLength(len(b))
+	}
+
+	if pk, ok := publicKeyCache.Get(cacheKey(b)); ok {
+		return pk.(*publicKey), nil
```

Comment:
I am asking you for double-checking my concern. You didn't copy the return of the LRU cache because the cache' return is not free in eviction or purge situations, right?

Reply:
Yes, I thought the reference counter should increase when the pointer is returned.
However, it seems nice to add Copy() as a precaution.